app/services/user_preferences/grantor.rb

```diff
@@ -0,0 +1,141 @@
+# frozen_string_literal: true
+
+# This class handles both the creating and updating of a user's UserPreferences.
+# It takes care of updating because, for each request, we are deleting all
+# of the user's associated UserPreference records, and creating new ones.
+
+# Also note that it accepts and sets multiple preference/choices combinations in
+# each request.  This is so that the end user can make a number of preference decisions
+# in one page, and the FE can send over just one request, in order to create all of the
+# UserPreference records for all of the users decisions.
+#
+# This class is responsible for:
+#   - accepting an Account record and an array of preference/choice pairings
+#   - destroying any associated existing UserPreference records
+#   - creating new UserPreference records based on the request
+#   - returning the successfully created pairings that were requested
+#
+module UserPreferences
+  class Grantor
+    attr_reader :account, :requested_user_preferences, :preference_codes, :user_preferences,
+                :preference, :preference_choice, :preference_records, :preference_choice_records
+
+    # @param account [Account] An instance of Account
+    # @param requested_user_preferences [Array<Hash>] An array of hashes. Each hash contains
+    #   a Preference#code, and the associates PreferenceChoice#codes the user selected.
+    #   For example:
+    #   [
+    #     {
+    #       preference: { code: 'benefits' },
+    #       user_preferences: [
+    #         { code: 'health-care' },
+    #         ...
+    #       ]
+    #     },
+    #     ...
+    #   ]
+    #
+    def initialize(account, requested_user_preferences)
+      @account = account
+      @requested_user_preferences = Validator.new(requested_user_preferences).of_presence!
+      @preference_codes = derive_preference_codes
+      @preference_records = eager_load_preferences
+      @preference_choice_records = eager_load_preference_choices
+    end
+
+    # @return [Array<Hash>] An array of hashes. Each hash contains the db Preference and
+    #   PreferenceChoice records that are associated with the user's newly created
+    #   UserPreference records.  For example:
+    #   [
+    #     {
+    #       preference: <Preference>,
+    #       user_preferences: [
+    #         <PreferenceChoice>,
+    #         <PreferenceChoice>,
+    #         ...
+    #       ]
+    #     },
+    #     ...
+    #   ]
+    #
+    def execute!
+      destroy_user_preferences!
```

Comment:
do we want to add a `rescue` block here in case something does go wrong we can report it Sentry?

Reply:
Addressed in https://github.com/department-of-veterans-affairs/vets-api/pull/2685/commits/ae111a98dc6dd951bad10d2c78a195ff9c86ebbf